### PR TITLE
Support 0.6

### PR DESCRIPTION
**Pull request by iamed2**
Adds some very specific version checks and fixes some (not all) deprecations.

### CODE REVIEW

File: src/executors.jl [hunk 1]
```diff
@@ -373,15 +381,26 @@ function dispatch!(exec::Executor, ctx::DispatchContext; throw_error=true)
...
-    f = Base.wrap_on_error(
-        Base.wrap_retry(
-            run_inner!,
-            allow_retry(retry_on(exec)),
-            retries(exec),
-            Base.DEFAULT_RETRY_MAX_DELAY
-        ),
-        on_error_inner!
-    )
+    @static if VERSION < v"0.6.0-dev.2042"
```
Comment (rofinn) [inline src/executors.jl]: Might be cleaner to do something like
```julia
retry_args = @static if VERSION < v"0.6.0-dev.2042"
    (retries(exec), Base.DEFAULT_RETRY_MAX_DELAY)
else
    ExponentialBackoff(; n=retries(exec))
end

f = wrap_on_error(wrap_retry(run_inner!, allow_retry(retry_on(exec)), retry_args...))
```

File: src/macros.jl [hunk 2]
```diff
@@ -129,7 +129,7 @@ function process_nodes!(ex::Expr, ctx_sym::Symbol)
     elseif ex.head === :dispatchinclude
         process_include!(ex, ctx_sym)
     else
-        map!(x->process_nodes!(x, ctx_sym), ex.args)
+        map!(x->process_nodes!(x, ctx_sym), ex.args, ex.args)
```
Comment (rofinn) [inline src/macros.jl:132]: These maps are kinda ugly. Do you know why `map!(f, collection)` was removed?

Comment (iamed2) [reply src/macros.jl:132]: https://github.com/JuliaLang/julia/pull/19721

File: src/executors.jl (hunk 1 above)
Comment (iamed2) [reply src/executors.jl]:
```julia
    retry_args = @static if VERSION < v"0.6.0-dev.2042"
        (allow_retry(retry_on(exec)), retries(exec), Base.DEFAULT_RETRY_MAX_DELAY)
    else
        (ExponentialBackOff(; n=retries(exec)), allow_retry(retry_on(exec)))
    end

    f = wrap_on_error(
        wrap_retry(
            run_inner!,
            retry_args...,
        ),
        on_error_inner!
    )
```

How's this?

Comment (rofinn) [reply src/executors.jl]: Yeah, that seems fine. It's annoying that the ordering of the arguments changed as well, but I think this clearly identifies what's actually different.

**Pull request merged**